### PR TITLE
paimon-cdc upgrade

### PR DESCRIPTION
**Pull request by newsuperchao**
1、Removed the JdbcUrlUtils class.
2、Added a configuration parameter 'scan.incremental.close-idle-reader.enabled' for MySQL CDC.
3、 upgrade flink-cdc to 2.4

### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/JdbcConstant.java [hunk 1]
```diff
@@ -0,0 +1,9 @@
+package org.apache.paimon.flink.action.cdc.mysql;
+
+/**
+ * @author jiangchao @Description:
+ * @date 2023/7/4 17:09
+ */
+public class JdbcConstant {
+    public static final String PROPERTIES_PREFIX = "jdbc.properties.";
```
Comment (JingsongLi) [inline paimon-flink/paimon-flink-common/src/main/java/org/apache/paimon/flink/action/cdc/mysql/JdbcConstant.java:8]: This field can just be in `MySqlActionUtils`.

**Review by JingsongLi — COMMENTED** [review]
Thanks @newsuperchao for the contribution, left one comment.

**Comment by yuzelin** [comment]
It seems something wrong. I have reported issue to flink-cdc community:
https://github.com/ververica/flink-cdc-connectors/issues/2337

**Comment by zhangjun0x01** [comment]
hi, @JingsongLi @yuzelin ,I check it and found It is indeed a bug for cdc 2.4 .it is caused by `scanNewlyAddedTables ` ,and I push a new PR ( https://github.com/apache/incubator-paimon/pull/1625 ),  all test cases are passed.

**Comment by zhuangchong** [comment]
I will close this PR, if you have any questions you can open this RP again.

**Pull request closed without merging**